src/vivarium_gates_nutrition_optimization_child/components/maternal_characteristics.py

```diff
@@ -272,11 +277,115 @@ def get_effect(self, index: pd.Index) -> pd.Series:
+
+    def __init__(self):
+        super().__init__(
+            "risk_factor.multiple_micronutrient_supplementation",
+            "risk_factor.gestational_age.birth_exposure",
+        )
+        self.excess_shift_pipeline_name = (
+            f"{self.risk.name}_on_{self.target.name}.excess_shift"
+        )
+        self.risk_specific_shift_pipeline_name = (
+            f"{self.risk.name}_on_{self.target.name}.risk_specific_shift"
+        )
+        self.raw_gestational_age_exposure_column_name = "raw_gestational_age_exposure"
+
+    #################
+    # Setup methods #
+    #################
+
+    # noinspection PyAttributeOutsideInit
+    def setup(self, builder: Builder) -> None:
+        super().setup(builder)
+        self.population_view = self._get_population_view(builder)
+        self.ifa_on_gestational_age = builder.components.get_component(
+            f"risk_effect.risk_factor.iron_folic_acid_supplementation.{self.target}"
+        )
+        self.mms_subpop_1_excess_shift = self._get_mms_excess_shift_data(
+            builder, data_keys.MMN_SUPPLEMENTATION.EXCESS_GA_SHIFT_SUBPOP_1
+        )
+        self.mms_subpop_2_excess_shift = self._get_mms_excess_shift_data(
+            builder, data_keys.MMN_SUPPLEMENTATION.EXCESS_GA_SHIFT_SUBPOP_2
+        )
+
+    def _get_population_view(self, builder: Builder) -> PopulationView:
+        return builder.population.get_view([self.raw_gestational_age_exposure_column_name])
+
+    def _get_mms_excess_shift_data(
+        self, builder: Builder, key: str
+    ) -> Dict[str, LookupTable]:
+        excess_shift_data = builder.data.load(
+            key, affected_entity=self.target.name, affected_measure=self.target.measure
+        )
+        excess_shift_data = self.build_excess_shift_lookup_table(builder, excess_shift_data)
+        return excess_shift_data
+
+    def build_excess_shift_lookup_table(
+        self, builder: Builder, excess_shift_data: pd.DataFrame
+    ) -> LookupTable:
+        """Reads excess shift data that was read from artifact and returns LookupTable with that data."""
+        excess_shift_data = rebin_relative_risk_data(builder, self.risk, excess_shift_data)
+        excess_shift_data = pivot_categorical(excess_shift_data)
+        return builder.lookup.build_table(
+            excess_shift_data, key_columns=["sex"], parameter_columns=["age", "year"]
+        )
+
+    ##################################
+    # Pipeline sources and modifiers #
+    ##################################
+    def _get_excess_shift_source(self, builder: Builder) -> Pipeline:
+        return builder.value.register_value_producer(
+            self.excess_shift_pipeline_name,
+            source=self.get_excess_shift,
+            requires_columns=[self.raw_gestational_age_exposure_column_name],
+        )
+
+    def _get_risk_specific_shift_source(self, builder: Builder) -> Pipeline:
+        return builder.value.register_value_producer(
```

Comment:
Do we have any modifiers on this pipeline? If not, why can't it just be a LookupTable which has 0 as its source?